File: test/test3.cc
#include "base.h"

void func(lua_State *L) {
	luaS_dofile(L, "test3.lua");
	LuaRef table = getGlobal(L, "table");
	assert(table[1].tostring() == "nil");
	table[1] = "c";
	assert(getGlobal(L, "newindex_msg").tostring() == "__newindex: 1, c");
	LuaRef e = table[1];
	assert(getGlobal(L, "index_msg").tostring() == "__index: 1");
	assert(e.tostring() == "nil");
	table[1] = table[2];
	assert(getGlobal(L, "index_msg").tostring() == "__index: 2");
	assert(getGlobal(L, "newindex_msg").tostring() == "__newindex: 1, nil");
}

File: test/test3.lua
table = setmetatable({}, {
	__index = function(t, k)
		index_msg =  '__index: ' .. tostring(k)
		return nil
	end,
	__newindex = function(t, k, v)
		newindex_msg = '__newindex: ' .. tostring(k) .. ', ' .. tostring(v)
	end,
})

--[[
a = table[1]
print(a)
table[1] = 0
a = table[1]
print(a)
--]]

